clients/spark/core/src/test/scala/io/treeverse/clients/ApiClientSpec.scala

```diff
@@ -32,17 +38,136 @@ class ApiClientSpec extends AnyFunSpec with Matchers {
     }
 
     describe("Azure") {
-      val translate = (u: String) => ApiClient.translateURI(new URI(u), StorageUtils.StorageTypeAzure).toString
+      val translate =
+        (u: String) => ApiClient.translateURI(new URI(u), StorageUtils.StorageTypeAzure).toString
       it("should translate any protocol to abfs") {
-        translate("https://account.example.net/container/path/to/blob") should be("abfs://container@account.dfs.core.windows.net/path/to/blob")
-        translate("ex://account.example.net/container/path/to/blob") should be("abfs://container@account.dfs.core.windows.net/path/to/blob")
+        translate("https://account.example.net/container/path/to/blob") should be(
+          "abfs://container@account.dfs.core.windows.net/path/to/blob"
+        )
+        translate("ex://account.example.net/container/path/to/blob") should be(
+          "abfs://container@account.dfs.core.windows.net/path/to/blob"
+        )
       }
 
       it("should handle empty paths") {
         // TODO(lynn): Trailing slashes might well be incorrect here.
-        translate("https://account.example.net/container/") should be("abfs://container@account.dfs.core.windows.net/")
-        translate("https://account.example.net/container") should be("abfs://container@account.dfs.core.windows.net/")
+        translate("https://account.example.net/container/") should be(
+          "abfs://container@account.dfs.core.windows.net/"
+        )
+        translate("https://account.example.net/container") should be(
+          "abfs://container@account.dfs.core.windows.net/"
+        )
       }
     }
   }
 }
+
+class RequestRetryWrapperSpec extends AnyFunSpec with Matchers with BeforeAndAfterEach {
```

Comment:
While I like how this test is implemented, it basically checks the functionality of an external library.
I think the test should be of the ApiClient class.

Reply:
I have not yet addressed this one, but I will add ApiClient tests too. I think that it is valuable to keep these tests too because it demonstrates how we expect the library to work. WDYT?

Reply:
Frankly, I don't see the value in it. The code should (and does) explain itself.

Reply:
I removed the unit tests. Invested this morning in trying to think of unit tests that could be useful, but did not have ideas. @johnnyaug will be happy to hear ideas if you have some!